sdk/cognitiveservices/azopenai/custom_client.go

```diff
@@ -109,17 +109,20 @@ func (b *openAIPolicy) Do(req *policy.Request) (*http.Response, error) {
 }
 
 // Methods that return streaming response
-
 type streamCompletionsOptions struct {
-	CompletionsOptions
+	// we strip out the 'stream' field from the options exposed to the customer so
+	// now we need to add it back in.
+	any
 	Stream bool `json:"stream"`
 }
 
 func (o streamCompletionsOptions) MarshalJSON() ([]byte, error) {
```

Comment:
Should MarshalJSON be exposed to customers?

Reply:
No, but Go gives us no choice in the matter - it must be exposed.